python/ray/serve/batching.py

```diff
@@ -445,33 +449,90 @@ def _validate_batch_wait_timeout_s(batch_wait_timeout_s):
         )
 
 
+SelfType = TypeVar("SelfType", contravariant=True)
```

Comment:
Why set `contravaraint=True` here?

Reply:
This is usually done for protocols where the input type is allowed to be any subtype of the given variable. [Here](https://peps.python.org/pep-0544/#generic-protocols) is a similar example from pep.

In our case, `SelfType` is used as a input parameter to the `__call__` method of the `_BatchDecorator` Protocol, so I hinted it to accept any subtype of the `SelfType`. This means that the decorated method is OK for use also if you subclass _BatchDecorator.